### PR TITLE
Fix DropDownButton with no items resulting in RenderFlex overflow

### LINKED ISSUE #26419: DropdownButtonFormField overflow issue when items are empty
I'm having overflow issue as following when I set items with empty list. 

dropdown_overflow_issue

## Steps to Reproduce

1. Run this snippet  https://gist.github.com/figengungor/446be330f5d34eb43ce6997feaf3b1ad

## Logs

```
```

```
```

```
[✓] Flutter (Channel beta, v0.11.10, on Mac OS X 10.14.1 18B75, locale en-TR)
    • Flutter version 0.11.10 at /Users/figengungor/Documents/development/flutter
    • Framework revision c27c4a265e (7 weeks ago), 2018-11-26 17:07:24 -0500
    • Engine revision eebc6a5895
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/figengungor/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 29.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] Connected device (1 available)
    • iPhone XR • 604D9667-5FAE-43A2-83D6-C3FA0EFF1307 • ios • iOS 12.1 (simulator)

```

### LINKED ISSUE #26875: DropdownButton overflow issue when onChanged is null
When `DropdownButton` is given a null argument for the `onChanged` parameter (or when no argument is present), a render error occurs with an infinite horizontal overflow.

## Steps to Reproduce

```dart
return Container(
  width: 500,
  height: 500,
  child: DropdownButton<String>(
    //onChanged: (str) {},   // This line fixes the problem
    value: 'abc',
    items: [
      DropdownMenuItem<String>(
        child: Text(
          'abc',
          overflow: TextOverflow.ellipsis,
        ),
        value: 'abc',
      ),
    ],
  ),
);
```

## Logs

```
I/flutter (26419): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (26419): The following assertion was thrown during performLayout():
I/flutter (26419): RenderIndexedStack object was given an infinite size during layout.
I/flutter (26419): This probably means that it is a render object that tries to be as big as possible, but it was put
I/flutter (26419): inside another render object that allows its children to pick their own size.
I/flutter (26419): The nearest ancestor providing an unbounded width constraint is:
I/flutter (26419):   RenderFlex#e51fb relayoutBoundary=up2 NEEDS-LAYOUT NEEDS-PAINT OVERFLOWING
I/flutter (26419):   creator: Row ← Padding ← Container ← DefaultTextStyle ← Stack ← Listener ← _GestureSemantics ←
I/flutter (26419):   RawGestureDetector ← GestureDetector ← Semantics ← DropdownButton<String> ← ConstrainedBox ← ⋯
I/flutter (26419):   parentData: offset=Offset(0.0, 0.0) (can use size)
I/flutter (26419):   constraints: BoxConstraints(0.0<=w<=500.0, 0.0<=h<=400.1)
I/flutter (26419):   size: Size(500.0, 400.1)
I/flutter (26419):   direction: horizontal
I/flutter (26419):   mainAxisAlignment: spaceBetween
I/flutter (26419):   mainAxisSize: min
I/flutter (26419):   crossAxisAlignment: center
I/flutter (26419):   textDirection: ltr
I/flutter (26419):   verticalDirection: down
I/flutter (26419): The constraints that applied to the RenderIndexedStack were:
I/flutter (26419):   BoxConstraints(0.0<=w<=Infinity, 0.0<=h<=400.1)
I/flutter (26419): The exact size it was given was:
I/flutter (26419):   Size(Infinity, 400.1)
I/flutter (26419): See https://flutter.io/layout/ for more information.
I/flutter (26419): 
I/flutter (26419): When the exception was thrown, this was the stack:
I/flutter (26419): #0      RenderBox.debugAssertDoesMeetConstraints.<anonymous closure> (package:flutter/src/rendering/box.dart:1749:9)
I/flutter (26419): #1      RenderBox.debugAssertDoesMeetConstraints (package:flutter/src/rendering/box.dart:1823:6)
I/flutter (26419): #2      RenderBox.size=.<anonymous closure> (package:flutter/src/rendering/box.dart:1543:17)
I/flutter (26419): #3      RenderBox.size= (package:flutter/src/rendering/box.dart:1543:65)
I/flutter (26419): #4      RenderStack.performLayout (package:flutter/src/rendering/stack.dart:480:7)
I/flutter (26419): #5      RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #6      RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
I/flutter (26419): #7      RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #8      RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:199:11)
I/flutter (26419): #9      RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #10     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
I/flutter (26419): #11     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #12     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #13     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #14     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #15     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #16     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #17     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #18     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:258:13)
I/flutter (26419): #19     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #20     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:142:11)
I/flutter (26419): #21     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:339:7)
I/flutter (26419): #22     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:212:7)
I/flutter (26419): #23     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:356:14)
I/flutter (26419): #24     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #25     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #26     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #27     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #28     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1188:11)
I/flutter (26419): #29     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #30     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #31     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #32     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #33     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #34     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #35     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #36     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #37     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #38     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #39     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #40     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #41     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #42     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #43     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #44     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #45     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:2881:13)
I/flutter (26419): #46     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #47     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
I/flutter (26419): #48     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #49     __RenderTheatre&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #50     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #51     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #52     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #53     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #54     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #55     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #56     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #57     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #58     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #59     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (26419): #60     RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (26419): #61     RenderView.performLayout (package:flutter/src/rendering/view.dart:147:13)
I/flutter (26419): #62     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1509:7)
I/flutter (26419): #63     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:768:18)
I/flutter (26419): #64     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:281:19)
I/flutter (26419): #65     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:677:13)
I/flutter (26419): #66     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:219:5)
I/flutter (26419): #67     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
I/flutter (26419): #68     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
I/flutter (26419): #69     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:751:7)
I/flutter (26419): #71     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
I/flutter (26419): #72     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
I/flutter (26419): #73     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
I/flutter (26419): (elided one frame from package dart:async)
I/flutter (26419): 
I/flutter (26419): The following RenderObject was being processed when the exception was fired:
I/flutter (26419):   RenderIndexedStack#6f903 relayoutBoundary=up3 NEEDS-LAYOUT NEEDS-PAINT
I/flutter (26419):   creator: IndexedStack ← Row ← Padding ← Container ← DefaultTextStyle ← Stack ← Listener ←
I/flutter (26419):   _GestureSemantics ← RawGestureDetector ← GestureDetector ← Semantics ← DropdownButton<String> ← ⋯
I/flutter (26419):   parentData: offset=Offset(0.0, 0.0); flex=null; fit=null (can use size)
I/flutter (26419):   constraints: BoxConstraints(0.0<=w<=Infinity, 0.0<=h<=400.1)
I/flutter (26419):   size: Size(Infinity, 400.1)
I/flutter (26419):   alignment: AlignmentDirectional.centerStart
I/flutter (26419):   textDirection: ltr
I/flutter (26419):   fit: loose
I/flutter (26419):   overflow: clip
I/flutter (26419):   index: null
I/flutter (26419): This RenderObject has no descendants.
I/flutter (26419): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (26419): Another exception was thrown: A RenderFlex overflowed by Infinity pixels on the right.
```

Flutter Doctor found no issues.

Versioning information:

```
[ +186 ms] [√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Version 10.0.17134.523], locale en-US)
[   +2 ms]     • Flutter version 1.0.0 at C:\src\flutter
[        ]     • Framework revision 5391447fae (8 weeks ago), 2018-11-29 19:41:26 -0800
[        ]     • Engine revision 7375a0f414
[        ]     • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

```

### PR DESCRIPTION
**Pull request by jimshepherd**
If the DropDownButton is provided a null or empty list to the items parameter, a RenderFlex overflow exception will be raised. This fix renders an empty Container instead of an IndexedStack to avoid the overflow. Unit test provided.

Fixes https://github.com/flutter/flutter/issues/26419
Fixes https://github.com/flutter/flutter/issues/26875

### CODE REVIEW

File: packages/flutter/test/material/dropdown_test.dart [hunk 1]
```diff
@@ -542,6 +542,34 @@ void main() {
...
   });
 
+  testWidgets('Size of DropdownButton with no items', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+    List<String> items;
+
+    Widget build() => buildFrame(buttonKey: buttonKey, items: items, onChanged: onChanged);
+
+    await tester.pumpWidget(build());
+    final RenderBox buttonBoxNullItems = tester.renderObject(find.byKey(buttonKey));
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: `tester.renderObject<RenderBox>` here and below

File: packages/flutter/test/material/dropdown_test.dart [hunk 2]
```diff
@@ -542,6 +542,34 @@ void main() {
...
+    assert(buttonBoxNullItems.attached);
+
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: please just leave one blank line between blocks of code, here and below

**Review by HansMuller — CHANGES_REQUESTED** [review]
This is a nice fix, sorry that it's taken so long to review it.

Just asking for some trivial changes here. When it's ready, I can land the PR.

File: packages/flutter/test/material/dropdown_test.dart [hunk 3]
```diff
@@ -542,6 +542,34 @@ void main() {
     expect(buttonBox.size, equals(buttonBoxNullValue.size));
   });
 
+  testWidgets('Size of DropdownButton with no items', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: Please insert this here
```
  // Regression test for https://github.com/flutter/flutter/issues/26419
```

File: packages/flutter/test/material/dropdown_test.dart (hunk 1 above)
Comment (jimshepherd) [reply packages/flutter/test/material/dropdown_test.dart]: Done

File: packages/flutter/test/material/dropdown_test.dart (hunk 2 above)
Comment (jimshepherd) [reply packages/flutter/test/material/dropdown_test.dart]: Done

**Review by HansMuller — APPROVED** [review]

**Comment by HansMuller** [comment]
Thanks for your contribution! It's been landed as is now running the final testing gauntlet. You can watch its status here: https://flutter-dashboard.appspot.com/build.html

**Pull request merged**